### PR TITLE
fix(cli/lib.deno.shared_globals): Move WorkerOptions::deno to unstable

### PR DESCRIPTION
**Pull request by nayeemrmn**

### CODE REVIEW

File: cli/dts/lib.deno.shared_globals.d.ts
Comment (nayeemrmn) [inline cli/dts/lib.deno.shared_globals.d.ts:711]: @bartlomieju It ended being quite complex because TS can't "merge" classes from two declaration files.

However, if we're willing to compromise and remove `WorkerOptions::deno: boolean` from stable, it would be an extremely minimal change due to the lack of property conflict. Just add the following in `lib.deno.unstable.d.ts`:
```ts
declare interface WorkerOptions {
  deno?: boolean | {
    namespace: boolean,
    permissions: { ... },
  }
}
```
Since it's already runtime unstable, I think zero user workflows would break...

Comment (bartlomieju) [reply cli/dts/lib.deno.shared_globals.d.ts:711]: @nayeemrmn sounds good, let's do this

Comment (bartlomieju) [reply cli/dts/lib.deno.shared_globals.d.ts:711]: Also, shouldn't the interface be exported?

**Review by bartlomieju — APPROVED** [review]
LGTM, waiting on Kitson's review before landing

**Pull request merged**
